### PR TITLE
Improvement: change `include_all_parents` default value of `TimeSeriesCausalGraph.extend_graph`

### PR DESCRIPTION
**Pull request by ilya-causalens**
## Description and Motivation

- The default value of the `include_all_parents` argument to `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.extend_graph`
  has been changed to `True` (from `False`).
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [ ] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [ ] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [ ] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

**Review by aslan-bakirov — APPROVED** [review]

File: cai_causal_graph/time_series_causal_graph.py [hunk 1]
```diff
@@ -435,16 +435,23 @@ def extend_graph(
         include nodes at time t-2, but rather that it will include nodes at time t-2 and all the nodes that are
         connected to them as specified by the minimal graph.
 
+        By default, this will also add all parents of newly added nodes. This is done to ensure that all nodes
+        corresponding to the same variable name have consistent parents, which is particularly useful for
+        causal modelling tasks. This can be controlled by the `include_all_parents` parameter. If set to `False`, then
```
Comment (andrew-causalens) [inline cai_causal_graph/time_series_causal_graph.py:440]: modelling -> modeling

File: changelog.md [hunk 2]
```diff
@@ -20,6 +23,8 @@
 - Generalized `cai_causal_graph.causal_graph.CausalGraph.__eq__` to check for the class of the instance itself, enabling
   to reuse this method by extending classes.
 - Dropped support for `python` `3.8` as it is approaching end of life.
+- The default value of the `include_all_parents` argument to `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.extend_graph`
+  has been changed to `True` (from `False`).
```
Comment (andrew-causalens) [inline changelog.md:27]: Would move this higher up. At least above the dropping of 3.8 support.

File: docs/time_series_causal_graph.md [hunk 3]
```diff
@@ -305,10 +305,16 @@ is obtained by extending (in time) the minimal graph with all the edges to the c
...
 `forward_steps`. If a backward step of `n` is specified, it means that the graph will be extended in order to
-include nodes back to time `-n` and all the nodes connected to them as specified by the minimal graph. An extra 
-`include_all_parents` flag can be set so that, in addition to all nodes as far back as `backward_steps`, extra nodes 
-and edges will be added as far back that all nodes up to `backward_steps` in the past have all their parents and 
-inbound edges. This means that the extended graph may now have nodes at lags further back than `backward_steps`.
+include nodes back to time `-n` and all the nodes connected to them as specified by the minimal graph. 
+
+By default, in addition to all nodes as far back as `backward_steps`, extra nodes and edges will be added as far back that all nodes 
+up to `backward_steps` in the past have all their parents and inbound edges. This means that the extended graph may now 
+have nodes at lags further back than `backward_steps`. This behaviour ensures that by default, all nodes for the same
```
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:312]: behaviour -> behavior

File: docs/time_series_causal_graph.md [hunk 4]
```diff
@@ -305,10 +305,16 @@ is obtained by extending (in time) the minimal graph with all the edges to the c
...
+variable will have consistent parents, i.e. if a node for variable `'X'` at time `n` has a parent `'Y'` at time `k`, then
+a newly added node `'X'` at time `n - j` will have a parent `'Y'` at time `k - j`. This is especially important for
+causal modeling tasks, such as Structural Causal Models. This behaviour can, however, be disabled by passing
```
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:315]: behaviour -> behavior

File: docs/time_series_causal_graph.md [hunk 5]
```diff
@@ -321,10 +327,11 @@ ts_cg: TimeSeriesCausalGraph
...
 forward_steps = 3
 
+# With all parents
 extended_graph = ts_cg.extend_graph(backward_steps=backward_steps, forward_steps=forward_steps)
 
-# With all parents
-extended_graph = ts_cg.extend_graph(backward_steps=backward_steps, forward_steps=forward_steps, include_all_parents=True)
+# Without all parents
+extended_graph = ts_cg.extend_graph(backward_steps=backward_steps, forward_steps=forward_steps, include_all_parents=False)
 ```
```
Comment (andrew-causalens) [inline docs/time_series_causal_graph.md:336]: I can't comment on the line but need to update the `## Example` section below.  It currently has
```
If you extend the minimal graph `backward_steps=2` and `forward_steps=0` with default arguments:

![ts_cg](images/ts_extended_graph.png)

Finally, if you extend the minimal graph with `backward_steps=2` and `forward_steps=0` but setting 
`include_all_parents=True`:

![ts_cg](images/ts_extended_graph_all_parents.png)
```
So just need to flip it essentially. Can see what it currently looks like [here](https://causalgraph.causalens.com/docs/generated/cai-causal-graph/docs/time_series_causal_graph#example).

**Review by andrew-causalens — COMMENTED** [review]
One piece of docs still needs updating. Rest is just US vs UK spelling. I won't approve as it doesn't matter anyways as I think the approval gets dismissed with new commits anyways.

**Review by andrew-causalens — APPROVED** [review]

**Pull request merged**